### ISSUE TITLE
[3.0]: Post editor ignores "Show subjects in topic"

### ISSUE THREAD

**Issue by LexArma** [SimpleMachines/SMF#8323]
### Basic Information

Small thing, but if the admin has hidden reply subjects in topic, perhaps the editor should not allow editing them either. 
Currently editing the subject line of a reply allows for semi-hidden added content, which can be hard to spot for moderators/admin.
Suggestion, if "Show subjects in topic" is disabled, simply use the default re: subject and don't allow editing it.

### Steps to reproduce

Toggle Admin -> Forum -> Posts and Topics -> Topic Settings -> Show subjects in topics, and post a new reply to see how the editor behaves. 

### Expected result

The editor should IMHO, respect the setting and allow/disallow editing the subject line accordingly.

### Actual result

Currently the editor always allows editing the subject.

### Version/Git revision

2.1.4

### Database Engine

All

**Comment by jdarwood007** [comment]
I think this is a candidate for 3.0, not 2.1.  The reason is that the change is to the template, and I don't consider this a bug.
Anyone else have a problem with moving this to 3.0?

**Comment by Sesquipedalian** [comment]
> I think this is a candidate for 3.0, not 2.1. The reason is that the change is to the template, and I don't consider this a bug. Anyone else have a problem with moving this to 3.0?

Agreed.

**Comment by LexArma** [comment]
I won't fight you guys, but would like to point out that we've seen spammers utilizing this, hiding some nasty keyword spam in topic titles so that they pop up in places (Like recent posts and RSS feeds), but are nowhere to be seen inside the actual topic. I consider that a bug. It's an actual risk, that might get a site blocked on services like Blue Coat.

**Comment by Sesquipedalian** [comment]
Hm. That is a fair point. Still, I am very reluctant to make theme changes in patches. We have a general policy against it for good reason.

Perhaps this issue could be addressed in another way. As I type this, I have the beginnings of a couple of ideas starting to form. I'll ponder and experiment over the next little while to see whether those ideas go anywhere.

**Comment by jdarwood007** [comment]
A solution could be if the setting is checked, we override the subject on submitted replies to force them to be "Re: $subject"

**Comment by LexArma** [comment]
Yes, that could work just fine. Could confuse some people, but would solve the actual issue.

**Comment by Sesquipedalian** [comment]
That's the main part of what I was thinking, yes. I also wanted to see if I could find a way to safely change the UI so that the subject field isn't even shown in these circumstances—most likely by using some JavaScript shenanigans.
